src/client/Sidebar/Item.jsx

```diff
@@ -8,6 +14,52 @@ export const Item = ({
 }) => {
   const [isHovered, setIsHovered] = useState(false);
 
+  const [showModal, setShowModal] = useState(false);
+
+  //Function to open the modal
+  const handleModalOpen = useCallback(() => {
+    setShowModal(true);
+  }, []);
+
+  //Function to close the modal
+  const handleModalClose = useCallback(() => {
+    setShowModal(false);
+  }, []);
+
+  //function to handle confirmation on modal
+  const handleConfirmModal = useCallback(() => {
+    setShowModal(false);
+    handleDeleteClick();
+  }, [handleDeleteClick]);
+
+  //function to close Modal if anywhere else on the screen is clicked
+  const outsideClickHandler = useCallback((ref) => {
+    useEffect(() => {
```

Comment:
Nope. This is odd. One of the "rules of hooks" is not to invoke hooks inside hooks. They should all be at the same level. See https://react.dev/warnings/invalid-hook-call-warning

In this case we can just remove the useCallback here.